pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -167,6 +167,13 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 		return s.Message(), ErrCreateContainerConfig
 	}
 
+	err = m.internalLifecycle.PreCreateContainer(pod, container, containerConfig)
+	if err != nil {
+		s, _ := grpcstatus.FromError(err)
+		m.recordContainerEvent(pod, container, "", v1.EventTypeWarning, events.FailedToCreateContainer, "Internal PreCreateContainer hook failed: %v", s.Message())
+		return s.Message(), ErrPreStartHook
```

Comment:
Is `ErrPreStartHook` the correct error type to return here? Should we add a new `ErrPreCreateHook` error?

Reply:
True, done